pkg/destroy/destroyer.go

```diff
@@ -0,0 +1,78 @@
+package destroy
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	atd "github.com/openshift/hive/contrib/pkg/aws_tag_deprovision"
+	log "github.com/sirupsen/logrus"
+
+	"github.com/openshift/installer/pkg/asset/metadata"
+	"github.com/openshift/installer/pkg/types"
+)
+
+// Destroyer allows multiple implementations of destroy
+// for different platforms.
+type Destroyer interface {
+	Run() error
+}
+
+// NewDestroyer returns Destroyer based on `metadata.json` in `rootDir`.
+func NewDestroyer(level log.Level, rootDir string) (Destroyer, error) {
```

Comment:
I'd rather this took a `*metadata.Metadata` with the file loading and deserialization happing externally (or in a `NewDestroyerFromFile` helper, or some such).  That would allow folks to create destroyers without hitting the disk (e.g. if they pulled metadata out of a cluster asset).